jerry-core/parser/js/js-parser-internal.h

```diff
@@ -118,9 +118,9 @@ typedef struct
  * All data allocated by the parser is
  * stored in parser_data_pages in the memory.
  */
-typedef struct parser_mem_page_t
+typedef struct parser_mem_page_s
```

Comment:
I'm open to discussions. I've seen quite some opinion wars "out there", I surely don't want to start one. If there are more voting for `typedef x_t { ... } x_t;` then I can drop this. (Or just keep that part of the patch that removes the struct tags where not needed. Or... add struct tags everywhere. Really, the key motivation for this PR was consistency throughout the code base.)